### PR TITLE
asset resolver balancer

### PR DESCRIPTION
**Pull request by revomhere**

### CODE REVIEW

File: lib/resolver/protocolAssets/resolver/balancer.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+                address: gauge,
+                abi: abiGauge,
+                functionName: "lp_token",
+                chainId,
+                args: [],
+            }
+        })
+    }) as (string | null)[]
+
+    return lpTokens.filter(lpToken => !!lpToken)
```
Comment (RedVeil) [inline lib/resolver/protocolAssets/resolver/balancer.ts:54]: why is this required?

Comment (revomhere) [reply lib/resolver/protocolAssets/resolver/balancer.ts:54]: here's some contracts don't have `lp_token` method, and if i call them - i will have `null`, so i should filter them

Comment (RedVeil) [reply lib/resolver/protocolAssets/resolver/balancer.ts:54]: ah right those are the side chain gauges, makes sense
